### PR TITLE
KREST-11276 - Tighten SSL auto cert reload

### PR DESCRIPTION
**Pull request by AndrewJSchofield**
Flaky test for SSL cert auto reload seems to be failing because the code itself isn't rock-solid :)

### CODE REVIEW

File: core/src/main/java/io/confluent/rest/SslFactory.java [hunk 1]
```diff
@@ -50,12 +50,12 @@ public static SslContextFactory createSslContextFactory(SslConfig sslConfig) {
           FileWatcher.onFileChange(watchLocation, () -> {
                 // Need to reset the key store path for symbolic link case
                 sslContextFactory.setKeyStorePath(sslConfig.getKeyStorePath());
-                sslContextFactory.reload(scf -> log.info("Reloaded SSL cert"));
+                sslContextFactory.reload(scf -> log.info("SSL cert auto reload begun"));
               }
```
Comment (omkreddy) [inline core/src/main/java/io/confluent/rest/SslFactory.java:54]: Can we add a log after SSL reload completes

Comment (AndrewJSchofield) [reply core/src/main/java/io/confluent/rest/SslFactory.java:54]: Couldn't figure out how to do it.

Comment (omkreddy) [reply core/src/main/java/io/confluent/rest/SslFactory.java:54]: Can't we add a log after `sslContextFactory.reload` line

Comment (AndrewJSchofield) [reply core/src/main/java/io/confluent/rest/SslFactory.java:54]: Is it a properly synchronous call with all side-effects guaranteed to have been completed by the time it returns? Why does the reload method take a `Consumer` if it's actually no use at all? I will add a log line. I doubt it is really that informative in practice.

**Review by trnguyencflt — APPROVED** [review]

**Review by msn-tldr — APPROVED** [review]

**Pull request merged**